tensorflow/python/keras/layers/wrappers.py

```diff
@@ -304,16 +351,19 @@ def compute_mask(self, inputs, mask=None):
         output_mask = K.any(output_mask, axis=-1)
     else:
       # output_mask is not None. We need to reshape it
-      input_length = input_shape[1]
+      input_length = tf_utils.convert_shapes(input_shape)
+      input_length = nest.flatten(input_length)[1]
       if not input_length:
-        input_length = K.shape(inputs)[1]
+        input_length = nest.map_structure(lambda x: K.shape(x)[1], inputs)
+        input_length = generic_utils.to_list(input_length)[0]
```

Comment:
Following your comment on line 172, this should also be changed to: `nest.flatten(is_ragged_input)`.